Dev 2021

lbednarczyk opened the pull request:
Updates 2021.

lbednarczyk committed:
Switched tab layout and added title to overview

lbednarczyk committed:
Updated Q-Q plot link

lbednarczyk committed:
Formatting

lbednarczyk committed:
Formatting

lbednarczyk committed:
Removed home and info buttons from explore and game pages

lbednarczyk committed:
Fixed buttons and added reference page

lbednarczyk committed:
Fixed view feedback for first challenge

lbednarczyk committed:
Formatting

lbednarczyk committed:
Fixed view feedback button

lbednarczyk committed:
combined ui and server

lbednarczyk committed:
Combined ui and server

lbednarczyk committed:
Got app.R running

lbednarczyk committed:
Renamed

lbednarczyk committed:
Combined into app.R

lbednarczyk committed:
Fixed typos

lbednarczyk committed:
Added description file

lbednarczyk committed:
Updated formatting of question bank

lbednarczyk committed:
Updating formatting in ui

lbednarczyk committed:
Added feedback

lbednarczyk committed:
Updated question bank formatting

lbednarczyk committed:
Added alt text to images.

lbednarczyk committed:
Added feedback

lbednarczyk committed:
Formatting

lbednarczyk committed:
Formatting and made code explicit

lbednarczyk committed:
Deleted unused packages

lbednarczyk committed:
Updated references

lbednarczyk committed:
Updated description

lbednarczyk committed:
Transitioned to app.R

lbednarczyk committed:
Updated folders

lbednarczyk committed:
Added packages

Jiayue-He commented:
Hi Lydia, I've reviewed your code and it was good and very neat. I can run the app but got some warnings when I generated the plots. I tried to google it but did not find a way to fix it.

Jiayue-He commented on app.R:
```diff
@@ -0,0 +1,1531 @@
+    
+    #adding pages to sidebar ----
+    dashboardSidebar(
+      width = 220,
```
Would it be 250 instead of 220 according to the study guide?

Jiayue-He commented on app.R:
```diff
@@ -0,0 +1,1531 @@
+          logx2 <-log(nonex2)
+          logy <- log(y)
+          
+          expx <- sqrt(nonex)
```
Here I got some warnings of NAs, I think it should be fine

paridhi1603 reviewed (commented):
Looks good to me after all the changes Jiayue suggested

lbednarczyk committed:
Formatting changes

lbednarczyk committed:
Updated explore page

lbednarczyk committed:
Fixed hints on explore page

neilhatfield commented:
@paridhi1603 @Jiayue-He Please see the recent message in Teams for how to register your approving reviews and then do so. (This will clear a system block.)

paridhi1603 commented:
Approved this pull request

On Thu, Jul 8, 2021 at 1:54 PM Neil Hatfield ***@***.***>
wrote:

> @paridhi1603 <https://github.com/paridhi1603> @Jiayue-He
> <https://github.com/Jiayue-He> Please see the recent message in Teams for
> how to register your approving reviews and then do so. (This will clear a
> system block.)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/EducationShinyAppTeam/Assumptions/pull/10#issuecomment-876632001>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANNLPDI23RZBUNSRDERX3MDTWXQ4TANCNFSM46BKG3UQ>
> .
>


-- 
Sincerely,
*Paridhi Khandelwal*
*The Pennsylvania State University | Class of 2022*
*Computational Data Science | College of Engineering*

neilhatfield reviewed (changes requested):
@lbednarczyk Here are a few changes:

1. README.md file
    + Go ahead and change the lifecycle badge from questioning to experimental; See [Badges](https://educationshinyappteam.github.io/Style_Guide/template.html#badges)
2. DESCRIPTION file
    + Update the date to be when you make this set of changes
3. Watch out for long lines (more than 80 characters) for both lines of code and comments (e.g., line 10). You can make a multi-line comment by breaking the line and putting the pound sign at the front of the following lines
4. What's the difference between `bank.csv` and `ChallengeOutput.csv`?
    + Let's try to come up with some better names for at least one of these files.
    + For both of these, load them just after `GRID_SIZE` and `TILE_COUNT` are defined and just before the ui starts; this will help with resource sharing.
5. Existing Issues--which of the registered issues have been fixed?
    + For example #2, #5, #7, and #8 
    + The new issue #11
6. Is the `shinyjs` package actually necessary? I've been running the app successfully without the package. Test and find out for sure one way or the other.
7. In the `www` directory, delete any images or other files which are no longer needed. For example, `Feature.css`, `delete.md`, `O.PNG`, etc.
8. Code Formatting--unless you can fit all of the arguments on the same line, the first argument will go down one line. Thus instead of how you currently have lines 17 & 18, these would look like the following:
    ```
    dashboardHeader(
      title = "Regression Assumptions and Diagnostics",
      titleWidth = 250,
      [etc.]
    )
    ```
    This holds true throughout the code with only a couple of exceptions like the header buttons and the `menuItems` of the `sidebarMenu`--both of these can put as long lines if you so choose.
9. The title in the dashboard header is too long; the "and Diagnostics" are cut off and not displaying. Go ahead and trim to just "Regression Assumptions" as the short title.
10. In line 73, go ahead and use the full title if that is "Regression Assumptions and Diagnostics".
11. Lines 86-105: these list items are not enclosed in a [list environment](https://educationshinyappteam.github.io/Style_Guide/htmlCSS.html#lists)
12. Please put a semicolon just after "center" and just before the closing quotation mark in line 106, line 159, etc.
13.  Change `style = "danger"` to `style = "default"` in line 110.
    + Generally, buttons should be `style = "default"` unless there is a key reason to use a different style. The danger style works well for things like a Reset button. See [Button Style (Color)](https://educationshinyappteam.github.io/Style_Guide/commonElements.html#style-color) for more details.
    + Apply this comment to all buttons in the app.
14. Lines 124-157: see prior comment about lists
15. In Line 144 add `class = "bodylinks"` to the `tags$a` call and put spaces around the `=` with the `href`.
16. Explore page--This is going to take a bit of work
    + Remove all `sidebarLayout`, `sidebarPanel`, and `mainPanel` calls
    + The `fluidRow` and `column` calls are problematic--e.g., overly nested, which is what causes the app to not scale well with different sized windows.
    + I would aim for something like the following:
    ```
    fluidRow(
      column(
        width = 4,
        offset = 0,
        wellPanel(
          selectInput(
            inputId = "model",
            [etc.]
          )
          [etc. for rest of controls]
        ) # close well
      ), # close column
      column(
        width = 8,
        offset = 0,
        plotOutput("plots")
      )
    )
    ```
17. In general, I find the Explore page to be missing instructions. What is the user supposed to do and get out of this page?
18. We only use well panels (the shaded backgrounds) for controls. Display message do not get them.
19. If you had the freedom to redesign the Explore page, how would you do so? Use this as permission to change up this page.
20. All plots must have meaningful alt text. The default of "Plot object" IS NOT sufficient. See the resources listed in the [Style Guide-Alt Text](https://educationshinyappteam.github.io/Style_Guide/graphics.html#altText2) for suggestions on writing alt text
21. For the questions which have images, we need to get better quality images so that they are not blurry; their alt text also needs to be improved.

lbednarczyk committed:
Updated lifecycle badge

lbednarczyk committed:
Updated date

lbednarczyk committed:
Deleted unused pictures

lbednarczyk committed:
Changed the name of the question bank

lbednarczyk committed:
Aligned selected model with activities

lbednarczyk committed:
Updates

neilhatfield committed:
Fall 2021 Updates

This brings the app up to the point of putting in the new tic tac toe module. There will be an update to the repo name forthcoming.
